source/common/quic/client_connection_factory_impl.cc

```diff
@@ -71,6 +71,14 @@ createQuicNetworkConnection(Http::PersistentQuicInfo& info, Event::Dispatcher& d
       quic::QuicUtils::CreateRandomConnectionId(), server_addr, info_impl->conn_helper_,
       info_impl->alarm_factory_, quic_versions, local_addr, dispatcher, nullptr);
 
+  // Update config with latest srtt, if available.
+  if (rtt_cache.has_value()) {
+    quic::QuicServerId& server_id = info_impl->server_id_;
+    Http::AlternateProtocolsCache::Origin origin("https", server_id.host(), server_id.port());
+    std::chrono::microseconds rtt = rtt_cache.value().get().getSrtt(origin);
+    info_impl->quic_config_.SetInitialRoundTripTimeUsToSend(rtt.count());
```

Comment:
Should we guard this with an "if rtt is not 0" check?

Reply:
whoops, indeed